internal/pkg/action/operator_uninstall.go

```diff
@@ -69,62 +62,56 @@ func (u *OperatorUninstall) Run(ctx context.Context) error {
 	csv, csvName, err := u.getSubscriptionCSV(ctx, sub)
 	if err != nil && !apierrors.IsNotFound(err) {
 		if csvName == "" {
-			return fmt.Errorf("get subscription CSV: %v", err)
+			return fmt.Errorf("get subscription csv: %v", err)
 		}
-		return fmt.Errorf("get subscription CSV %q: %v", csvName, err)
+		return fmt.Errorf("get subscription csv %q: %v", csvName, err)
 	}
 
-	// Deletion order:
-	//
-	// 1. Subscription to prevent further installs or upgrades of the operator while cleaning up.
-	// 2. CustomResourceDefinitions so the operator has a chance to handle CRs that have finalizers.
-	// 3. ClusterServiceVersion. OLM puts an ownerref on every namespaced resource to the CSV,
-	//    and an owner label on every cluster scoped resource so they get gc'd on deletion.
+	// find operands related to the operator on cluster
+	lister := action.NewOperatorListOperands(u.config)
+	operands, err := lister.Run(ctx, u.Package)
+	if err != nil {
+		return fmt.Errorf("list operands for operator %q: %v", u.Package, err)
+	}
+	// validate the provided deletion strategy before proceeding to deletion
+	if ok, err := u.validStrategy(operands); !ok {
+		return fmt.Errorf("could not proceed with deletion of %q: %s", u.Package, err)
+	}
```

Comment:
Nit: It isn't possible with the current implementation, but if `validStrategy()` returns `false, nil` we'll get a strange looking error message.

Perhaps we can just make this function return just an `error`?
```suggestion
	if err := u.validateStrategy(operands); err != nil {
		return fmt.Errorf("could not proceed with deletion of %q: %s", u.Package, err)
	}
```